### PR TITLE
Widget Visibility: support custom post type archive pages #80

### PR DESCRIPTION
**Pull request by jessefriedman**
fixes for #80 and #750
replaces #1599

### CODE REVIEW

File: modules/widget-visibility/widget-conditions.php [hunk 1]
```diff
@@ -220,6 +251,7 @@ public static function widget_conditions_admin( $widget, $return, $instance ) {
 									<option value="tag" <?php selected( "tag", $rule['major'] ); ?>><?php echo esc_html_x( 'Tag', 'Noun, as in: "This post has one tag."', 'jetpack' ); ?></option>
 									<option value="date" <?php selected( "date", $rule['major'] ); ?>><?php echo esc_html_x( 'Date', 'Noun, as in: "This page is a date archive."', 'jetpack' ); ?></option>
 									<option value="page" <?php selected( "page", $rule['major'] ); ?>><?php echo esc_html_x( 'Page', 'Example: The user is looking at a page, not a post.', 'jetpack' ); ?></option>
+									<option value="post_type" <?php selected( "post_type", $rule['major'] ); ?>><?php echo esc_html_x( 'Post Type', 'Show/Hide if user in certain custom post type archive.', 'jetpack' ); ?></option>
```
Comment (eliorivero) [inline modules/widget-visibility/widget-conditions.php]: This string "Show/Hide if user in certain custom post type archive." should be "Show/Hide if user is in a certain custom post type archive."

**Comment by eliorivero** [comment]
Works good. Needs update to solve conflicts.

**Comment by jeherve** [comment]
Rebased.

File: modules/widget-visibility/widget-conditions.php [hunk 2]
```diff
@@ -269,7 +300,7 @@ public static function widget_conditions_admin( $widget, $return, $instance ) {
 									<option value="tag" <?php selected( "tag", $rule['major'] ); ?>><?php echo esc_html_x( 'Tag', 'Noun, as in: "This post has one tag."', 'jetpack' ); ?></option>
 									<option value="date" <?php selected( "date", $rule['major'] ); ?>><?php echo esc_html_x( 'Date', 'Noun, as in: "This page is a date archive."', 'jetpack' ); ?></option>
 									<option value="page" <?php selected( "page", $rule['major'] ); ?>><?php echo esc_html_x( 'Page', 'Example: The user is looking at a page, not a post.', 'jetpack' ); ?></option>
-
+									<option value="post_type" <?php selected( "post_type", $rule['major'] ); ?>><?php echo esc_html_x( 'Post Type', 'Show/Hide if user is in a certain custom post type archive.', 'jetpack' ); ?></option>
```
Comment (dereksmart) [inline modules/widget-visibility/widget-conditions.php]: This is a very strange Translator Context comment

Comment (jeherve) [inline modules/widget-visibility/widget-conditions.php]: Indeed. Do we really need a comment there? "Post Type" seems pretty standard.

Comment (eliorivero) [inline modules/widget-visibility/widget-conditions.php]: I agree. I've modified it following the wording of the one used for `Page`

**Comment by zinigor** [comment]
Tested, works well. I have adjusted the text, as well as refactored the markup a bit.

**Pull request merged**